core/shared/src/main/scala/org/scalacheck/derive/Instances.scala

```diff
@@ -140,3 +141,62 @@ trait EnumerationInstances {
    ): Arbitrary[E#Value] =
     Arbitrary(Gen.oneOf(w.value.values.toSeq))
 }
+
+trait TaggedInstances {
+  def arbitraryTagged[U: Arbitrary, T]: Arbitrary[U @@ T] =
```

Comment:
Doesn't it work when making this one implicit, and removing the other `Arbitrary` instances?

Reply:
I tried that but under some circumstances this yields a stack overflow error in the compiler. This approach is more verbose, but seems to work always.

Reply:
Sorry in which compiler version do you see the SO? This should work.